velox/exec/RowContainer.cpp

```diff
@@ -234,23 +245,28 @@ RowContainer::RowContainer(
   // no nulls, it may be that there are no null flags.
   if (!nullOffsets_.empty()) {
     // All flags like free and probed flags and null flags for keys and non-keys
-    // start as 0.
-    initialNulls_.resize(nullBytes, 0x0);
-    // Aggregates are null on a new row.
-    const auto aggregateNullOffset = nullableKeys ? keyTypes.size() : 0;
-    for (int32_t i = 0; i < accumulators_.size(); ++i) {
-      bits::setBit(initialNulls_.data(), i + aggregateNullOffset);
-    }
+    // start as 0. This is also used to mark aggregates as uninitialized on row
+    // creation.
+    initialNulls_.resize(flagBytes_, 0x0);
   }
   originalNormalizedKeySize_ = hasNormalizedKeys_
       ? bits::roundUp(sizeof(normalized_key_t), alignment_)
       : 0;
   normalizedKeySize_ = originalNormalizedKeySize_;
+  size_t nullOffsetsPos = 0;
   for (auto i = 0; i < offsets_.size(); ++i) {
     rowColumns_.emplace_back(
         offsets_[i],
-        (nullableKeys_ || i >= keyTypes_.size()) ? nullOffsets_[i]
+        (nullableKeys_ || i >= keyTypes_.size()) ? nullOffsets_[nullOffsetsPos]
                                                  : RowColumn::kNotNullOffset);
+
+    if (!accumulators.empty() && i >= keyTypes_.size() &&
```

Comment:
in which case i could >= keyTypes_.size() + accumulators.size()? Thanks!

Reply:
The layout for the null bits is those for keys, then those for accumulators, then those for dependent types.

So if i is pointing to the null bit for a dependent type it could be greater than the size of keyTypes and accumulators.

I'll add a comment.